tests/test_filter_job_by_email.py

```diff
@@ -0,0 +1,34 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from pages.treeherder import TreeherderPage
+
+
+@pytest.mark.nondestructive
+def test_filter_by_email(base_url, selenium):
+    """Open Treeherder page, select and filter by random email,
+    verify if filtered emails have the same email address"""
+    page = TreeherderPage(selenium, base_url).open()
+    page.select_random_email()
+
+    filtered_emails_name = page.result_sets[0].email_name
+    random_email_name = page.random_email_name
+
+    assert filtered_emails_name == random_email_name
+
+
+@pytest.mark.nondestructive
+def test_remove_email_address_filter(base_url, selenium):
+    """Open Treeherder page, select and filter by random email,
+    remove email address filter, verify that job list display all authors"""
+    page = TreeherderPage(selenium, base_url).open()
+    page.select_random_email()
+
+    page.click_on_active_watched_repo()
+    first_email = page.random_email_name
+    second_email = page.random_email_name
+
+    assert first_email != second_email
```

Comment:
I just loaded Treeherder and there was one email address listed for multiple jobs- so while it's improbable, it's not impossible that 2 randomly selected emails will be for the same address. Perhaps it would be better to assert that the all_emails list contains more than just the first email found.